File: C++/length-of-the-longest-alphabetical-continuous-substring.cpp
// Time:  O(n)
// Space: O(1)

// string
class Solution {
public:
    int longestContinuousSubstring(string s) {
        int result = 0;
        for (int i = 0, l = 0; i < size(s); ++i) {
            ++l;
            if (i + 1 == size(s) || s[i] + 1 != s[i + 1]) {
                result = max(result, l);
                l = 0;
            }
        }
        return result;
    }
};

File: Python/length-of-the-longest-alphabetical-continuous-substring.py
# Time:  O(n)
# Space: O(1)

# string
class Solution(object):
    def longestContinuousSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        result = l = 0
        for i in xrange(len(s)):
            l += 1
            if i+1 == len(s) or ord(s[i])+1 != ord(s[i+1]):
                result = max(result, l)
                l = 0
        return result
